<!-- THIS LAUNCH FILE WAS AUTO-GENERATED FROM, DO NOT MODIFY src/auv_config.py -->

<launch>
	<arg name="mission_complete_topic" default="mission_complete" />
	<arg name="robot_name" default="sam" />
	<arg name="emergency_topic" default="abort" />
	<arg name="plan_control_topic" default="imc/plan_control" />
	<arg name="dvl_cooldown" default="0.5" />
	<arg name="depth_topic" default="ctrl/odom_listener/depth_feedback" />
	<arg name="dvl_running_depth" default="0.55" />
	<arg name="heartbeat_topic" default="heartbeat" />
	<arg name="leader_link" default="sam_1/base_link" />
	<arg name="absolute_min_altitude" default="1" />
	<arg name="utm_link" default="utm" />
	<arg name="start_stop_dvl_namespace" default="core/toggle_dvl" />
	<arg name="path_topic" default="ctrl/planned_path" />
	<arg name="vehicle_state_topic" default="imc/vehicle_state" />
	<arg name="base_link" default="sam/base_link" />
	<arg name="emergency_action_namespace" default="ctrl/emergency_surface_action" />
	<arg name="waypoint_tolerance" default="1.5" />
	<arg name="plan_control_state_topic" default="imc/plan_control_state" />
	<arg name="emergency_trials_before_giving_up" default="30" />
	<arg name="max_depth" default="20" />
	<arg name="min_altitude" default="5" />
	<arg name="abort_topic" default="imc/abort" />
	<arg name="action_namespace" default="ctrl/goto_waypoint" />
	<arg name="plan_viz_topic" default="viz/mission_waypoints" />
	<arg name="local_link" default="map" />
	<arg name="poi_detector_link" default="sam/camera_down_link" />
	<arg name="leak_topic" default="core/leak" />
	<arg name="plandb_topic" default="imc/plandb" />
	<arg name="altitude_topic" default="core/dvl" />
	<arg name="utm_band" default="None" />
	<arg name="path_planner_name" default="None" />
	<arg name="utm_zone" default="None" />
	<arg name="enable_leader_follower" default="True" />
	<arg name="min_distance_to_leader" default="5" />
	<arg name="gps_fix_topic" default="core/gps" />
	<arg name="follow_action_namespace" default="ctrl/leader_follower_action" />
	<arg name="camera_detection_topic" default="detection/poi_down" />
	<arg name="estimated_state_topic" default="imc/estimated_state" />


	<node name="sam_bt" pkg="smarc_bt" type="sam_bt.py" output="screen" ns="$(arg robot_name)">
		<param name="mission_complete_topic" value="$(arg mission_complete_topic)" />
		<param name="robot_name" value="$(arg robot_name)" />
		<param name="emergency_topic" value="$(arg emergency_topic)" />
		<param name="plan_control_topic" value="$(arg plan_control_topic)" />
		<param name="dvl_cooldown" value="$(arg dvl_cooldown)" />
		<param name="depth_topic" value="$(arg depth_topic)" />
		<param name="dvl_running_depth" value="$(arg dvl_running_depth)" />
		<param name="heartbeat_topic" value="$(arg heartbeat_topic)" />
		<param name="leader_link" value="$(arg leader_link)" />
		<param name="absolute_min_altitude" value="$(arg absolute_min_altitude)" />
		<param name="utm_link" value="$(arg utm_link)" />
		<param name="start_stop_dvl_namespace" value="$(arg start_stop_dvl_namespace)" />
		<param name="path_topic" value="$(arg path_topic)" />
		<param name="vehicle_state_topic" value="$(arg vehicle_state_topic)" />
		<param name="base_link" value="$(arg base_link)" />
		<param name="emergency_action_namespace" value="$(arg emergency_action_namespace)" />
		<param name="waypoint_tolerance" value="$(arg waypoint_tolerance)" />
		<param name="plan_control_state_topic" value="$(arg plan_control_state_topic)" />
		<param name="emergency_trials_before_giving_up" value="$(arg emergency_trials_before_giving_up)" />
		<param name="max_depth" value="$(arg max_depth)" />
		<param name="min_altitude" value="$(arg min_altitude)" />
		<param name="abort_topic" value="$(arg abort_topic)" />
		<param name="action_namespace" value="$(arg action_namespace)" />
		<param name="plan_viz_topic" value="$(arg plan_viz_topic)" />
		<param name="local_link" value="$(arg local_link)" />
		<param name="poi_detector_link" value="$(arg poi_detector_link)" />
		<param name="leak_topic" value="$(arg leak_topic)" />
		<param name="plandb_topic" value="$(arg plandb_topic)" />
		<param name="altitude_topic" value="$(arg altitude_topic)" />
		<param name="utm_band" value="$(arg utm_band)" />
		<param name="path_planner_name" value="$(arg path_planner_name)" />
		<param name="utm_zone" value="$(arg utm_zone)" />
		<param name="enable_leader_follower" value="$(arg enable_leader_follower)" />
		<param name="min_distance_to_leader" value="$(arg min_distance_to_leader)" />
		<param name="gps_fix_topic" value="$(arg gps_fix_topic)" />
		<param name="follow_action_namespace" value="$(arg follow_action_namespace)" />
		<param name="camera_detection_topic" value="$(arg camera_detection_topic)" />
		<param name="estimated_state_topic" value="$(arg estimated_state_topic)" />
	</node>
</launch>
